test/testother.cpp

```diff
@@ -3885,6 +3886,68 @@ class TestOther : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void duplicateVarExpression() {
+        check("int f();\n"
+              "void test() {\n"
+              "    int i = f();\n"
+              "    int j = f();\n"
+              "}");
+        ASSERT_EQUALS("[test.cpp:4] -> [test.cpp:3]: (style) Both variables 'i' and 'j' are assigned the same expression.\n", errout.str());
```

Comment:
there is no indication that `f` returns the same value in this test. I would suggest that the function must be pure/const.

Reply:
I believe we'll need to consider carefully what expressions we can warn about before we apply this. I am not so sure about `++x` , `*p` , `abc.a`. I guess the warning is ok if there are heavy calculations in rhs.

Reply:
> there is no indication that f returns the same value in both calls in this test. I would suggest that the function must be pure/const.

True, but it still could be a copy/paste error. It might be better to make it inconclusive for non-const functions.

Reply:
> I am not so sure about ++x , *p , abc.a. I guess the warning is ok if there are heavy calculations in rhs.

We probably dont want to warn for `++x`. I dont see a problem with `*p` or `abc.a`.

Reply:
> I dont see a problem with *p or abc.a.

I agree it could be a copy/paste error but it might also be written in such way by intention.. We can try it on daca2 and see how it works.. if it finds real copy/paste bugs then it is great! If there are some specific expressions that are often repeated by intention then it should not warn about those.